### PR TITLE
Drop test_deps role

### PR DESCRIPTION
**Pull request by raukadah**
this role is already moved to ci-framework[1] and let'use it from there.

The existing one is outdated and breaking the existing zuul molecule job.

[1]. https://github.com/openstack-k8s-operators/ci-framework/tree/main/ci_framework/roles/test_deps

Depends-On: https://github.com/openstack-k8s-operators/ci-framework/pull/271

### CODE REVIEW

**Review by cjeanner — APPROVED** [review]
yay! convergence - more convergence!

**Comment by cjeanner** [comment]
Not that easy apparently - the container used for molecule doesn't seem to provide ci-framework roles, so test_deps isn't there, sooo... :crab:

**Review by jpodivin — CHANGES_REQUESTED** [review]
I don't really know about this. Moving all tests from github to rdo is, by itself, considerable change I'm not sure I would entirely embrace. But, and I feel this is more important, removing the role creates a dependency on ci-framework we don't really need here.

I've seen first hand where this can end up with tripleo ecosystem, where we had to make intertwined changes in multiple repos to address some CIX or another. And if we can spare ourselves that now, I think we should.

**Review by slagle — APPROVED** [review]

File: ci/playbooks/mol_role_symlink.yaml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+---
+- hosts: all
+  tasks:
+    - name: "Create {{ ansible_user_dir }}/.ansible/roles"
+      ansible.builtin.file:
+        path: "{{ ansible_user_dir }}/.ansible/roles"
+        state: directory
+        recurse: yes
+
+    - name: Create symlink for ci-framework test_deps role
```
[the remark this answers is not in the document]
Comment (raukadah) [reply ci/playbooks/mol_role_symlink.yaml:10]: Done

**Review by rebtoor — APPROVED** [review]

**Comment by raukadah** [comment]
recheck

**Review by slagle — APPROVED** [review]

**Comment by rebtoor** [comment]
> We have already wired ci-framework in edpm-ansible via makefile. Since we are already running zuul molecule jobs by using ci-framework . So I am extending the same work with other molecule jobs and re-using the role from ci-framework.

> We know both projects are dependent, since we are using Zuul, we can test the changes together. I think that is not an issue here.

On this point I want to add my 2 cents.

I totally agree on remove test_deps from edpm-ansible for a couple of reason:

1. you already have an updated version of it on ci-framework
2. i don't like the idea of having a role used only for testing purposes distributed in the collection. I don't remember see anything similar elsewhere either.

BTW, I'd like to keep execute molecule tests without triggering ci-framework. It shouldn't become a requirement for molecule. edpm-ansible should be a consumer of ci-framework just like the other repos in the organization. 

Anyone should have the possibility of cloning this repo and launch `molecule test` in a role folder without external dependencies.

To overcome that, we are thinking to write a "common" prepare phase for all the scenarios, adapting the amazing work started by @Deydra71 on https://github.com/openstack-k8s-operators/edpm-ansible/pull/171, that will replace the test_deps repo without using external tools.

I hope and I guess (given ansible idempotence) that this will not collide with test_deps role in the ci-framework.

I'll tag the rest of DF team to ask for their point of view. @slagle @bshephar @rabi @kajinamit

**Review by slagle — CHANGES_REQUESTED** [review]
I agree we should be able to run "molecule test" without having to rely on dependencies from ci-framework. Can we hold off on dropping this until we have the replacement in place that @rebtoor proposes?

**Pull request closed without merging**